src/models/streaming_server.rs

```diff
@@ -445,3 +514,32 @@ fn parse_torrent(torrent: &[u8]) -> Result<(String, Vec<String>), serde_bencode:
     announce.dedup();
     Ok((info_hash, announce))
 }
+
+fn play_on_device<E: Env + 'static>(url: &Url, args: &PlayOnDeviceArgs) -> Effect {
+    #[derive(Serialize)]
+    #[serde(rename_all = "camelCase")]
+    struct Body {
+        source: String,
+        time: u64,
+    }
+    let endpoint = url
+        .join(&format!("casting/{}/player", args.device))
```

Comment:
Could you give an example of the device argument? We should be certain that it does not need `urlencode` or always `urlencode` (which would be easier)

Reply:
Sure, you can check this action dispatch on stremio-web:
https://github.com/Stremio/stremio-web/blob/development/src/routes/Player/OptionsMenu/OptionsMenu.js#L52